### ISSUE TITLE
React import shouldn't be removed when using "Remove all unused imports" in jsx/tsx files

### ISSUE THREAD

**Issue by GeekCornerGH** [microsoft/TypeScript#56042]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.83.0-insider
- OS Version: Windows 11 Home 22621.2283

Steps to Reproduce:

1. Create a TSX file/JSX
2. Import React (`import React from "react"`)
3. Import another lib
4. Make sure none of the libs are referenced in the code
5. Ctrl + "." on the second lib
6. Click remove all unused imports
7. React is gone while it shouldn't

**Comment by RyanCavanaugh** [comment]
Whether or not this intended depends on your configuration; please post a complete sample

**Comment by GeekCornerGH** [comment]
> Hello!
> 
> I guess you're using an older (16.x) version of React? (Asking this because, if I remember well, from version 17, you don't need the React import anymore).

@bhubr I'm using react 18, and eslint requires React to be imported when using tsx

**Comment by GeekCornerGH** [comment]
> Whether or not this intended depends on your configuration; please post a complete sample

```ts
import React from "react";
import {config} from "dotenv";
```

Then proceed to resume from step 5

**Comment by bhubr** [comment]
@GeekCornerGH It's not a VSCode issue, but a matter of setting up ESLint.

I did encounter the issue many times before, when I chose "To check syntax and find problems" as an answer to the first question asked by the ESLint config tool.

I just created a quick Vite-based project, and when I setup ESLint with a style guide, the problem disappears, because it uses defaults more suited to modern React.

The in-depth explanation is in then [`react/react-in-jsx-scope` rule's documentation](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md), at the very bottom, under "When Not To Use It".

> If you are using the [new JSX transform from React 17](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports), you should disable this rule by extending [react/jsx-runtime](https://github.com/jsx-eslint/eslint-plugin-react/blob/8cf47a8ac2242ee00ea36eac4b6ae51956ba4411/index.js#L165-L179) in your eslint config (add "plugin:react/jsx-runtime" to "extends").

Since you're using React 18, you are indeed using the new JSX transforms, and should disable the rule.
